proc-macro/src/lib.rs

```diff
@@ -251,10 +251,10 @@ fn impl_trait_query(arg: TokenStream, item: TokenStream) -> Result<TokenStream2>
             }
         }
 
-        unsafe impl #impl_generics #imports::QueryData for &mut #trait_object
+        unsafe impl #impl_generics_with_lifetime #imports::QueryData for &'__a mut #trait_object
         #where_clause
         {
-            type ReadOnly = &'__a #trait_object;
+            type ReadOnly = &'__a mut #trait_object;
```

Comment:
Shouldn't this remain as &'__a mut?

Reply:
I mean it makes sense to me as well intuitively, but the compiler is not happy:

```
error[E0271]: type mismatch resolving `<&mut dyn Person as QueryData>::ReadOnly == &mut dyn Person`
   --> src/tests.rs:11:1
    |
11  | #[queryable]
    | ^^^^^^^^^^^^ types differ in mutability
    |
    = note: expected mutable reference `&mut _`
                       found reference `&_`
note: required by a bound in `ReadOnlyQueryData`
   --> /home/aviac/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_ecs-0.13.0/src/query/fetch.rs:273:47
    |
273 | pub unsafe trait ReadOnlyQueryData: QueryData<ReadOnly = Self> {}
    |                                               ^^^^^^^^^^^^^^^ required by this bound in `ReadOnlyQueryData`
    = note: this error originates in the attribute macro `queryable` (in Nightly builds, run with -Z macro-backtrace for more info)
```

---

Ahhh wait... this does need further adjustment, right?

Reply:
Mmh I think right below this there is an impl for ReadOnlyQueryData for `&mut #trait_object` which should be removed too